samples/web/content/apprtc/js/main.js

```diff
@@ -358,3 +399,142 @@ function toggleFullScreen() {
     trace(event);
   }
 }
+
+roomIdInput.addEventListener('input', function(){
+  // validate room id, enable/disable join button
+  // The server currently accepts only the \w character class
+  var room = roomIdInput.value;
+  var valid = room.length >= 5;
+  var re = /^\w+$/;
+  valid = valid && re.exec(room);
+  if (valid) {
+    joinRoomButton.disabled = false;
+  } else {
+    joinRoomButton.disabled = true;
+  }
+});
+
+randomRoomButton.addEventListener('click', function() {
+  roomIdInput.value = randomString(9);
+}, false);
+
+joinRoomButton.addEventListener('click', function() {
+  // TODO - validate entered room name
+  
+  roomId = roomIdInput.value;
+  loadRoom();
+},false);
+
+function showRoomSelectionUI(shouldShow) {
+  if (shouldShow) {
+    videosDiv.classList.add('hidden');
+    footerDiv.classList.add('hidden');
+    roomUIDiv.classList.add('active');
+  } else {
+    videosDiv.classList.remove('hidden');
+    footerDiv.classList.remove('hidden');
+    roomUIDiv.classList.remove('active');
+  }
+}
+
+if (!isChromeApp()) {
+  window.onpopstate = function(event) {
+    if (!event.state) {
+      // Resetting back to room selection page not yet supported, reload
+      // the initial page instead.
+      trace('Reloading main page.');
+      location.href = location.origin;
+    } else {
+      // This could be a forward request to open a room again
+      if (event.state && event.state.roomLink) {
+        location.href = event.state.roomLink;
+      }
+    }
+    
+  };
+}
+
+function loadRoom() {
+  pushRecentRoom(roomId).then(function() {
+    // check if room is available
+    var roomCheckUri = server + '/room/' + encodeURIComponent(roomId) + '/status';
```

Comment:
The room state might change after the /room/status response, so I don't think the room status check here adds much value. Why not just handle the "room full error" in the response of /register/?